File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/RandomBasedSnippetVariableResolver.java

package io.github.abc15018045126.sora.widget.snippet.variable;

import androidx.annotation.NonNull;

import java.util.Random;
import java.util.UUID;

public class RandomBasedSnippetVariableResolver implements ISnippetVariableResolver {

    @NonNull
    @Override
    public String[] getResolvableNames() {
        return new String[]{
                "RANDOM", "RANDOM_HEX", "UUID"
        };
    }

    @NonNull
    @Override
    public String resolve(@NonNull String name) {
        switch (name) {
            case "RANDOM":
                return Integer.toString(new Random().nextInt());
            case "RANDOM_HEX":
                return Integer.toString(new Random().nextInt(), 16);
            case "UUID":
                return UUID.randomUUID().toString();
        }
        throw new IllegalArgumentException("Unsupported variable name:" + name);
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/RandomBasedSnippetVariableResolver.kt
package io.github.abc15018045126.sora.widget.snippet.variable

import java.util.Random
import java.util.UUID

class RandomBasedSnippetVariableResolver : ISnippetVariableResolver {

    override fun getResolvableNames(): Array<String> {
        return arrayOf(
            "RANDOM", "RANDOM_HEX", "UUID"
        )
    }

    override fun resolve(name: String): String {
        return when (name) {
            "RANDOM" -> Random().nextInt().toString()
            "RANDOM_HEX" -> Random().nextInt().toString(16)
            "UUID" -> UUID.randomUUID().toString()
            else -> throw IllegalArgumentException("Unsupported variable name:$name")
        }
    }
}
